tests/neg/i7048e.scala

```diff
@@ -0,0 +1,39 @@
+import scala.quoted.{given, _}
+
+abstract class Test {
+  type T
+
+  val T: Type[T]
+  val getT: Type[T] = T // need this to avoid getting `null`
+  given Type[T] = getT
+
+  def foo with QuoteContext: Expr[Any] = {
+
+    val r = '{Option.empty[T]} // error
+
+    {
+      val t: Test = this
+      import t.{given}
+      println(summon[Type[t.T]].show)
+      // val r = '{Option.empty[t.T]} // access to value t from wrong staging level
+      val r2 = '{Option.empty[${t.T}]} // works
+    }
+
+    {
+      val r1 = '{Option.empty[${T}]} // works
+      val r2 = '{Option.empty[List[${T}]]} // works
+      // val r3 = '{summon[Type[${T}]]} // access to Test.this from wrong staging level
+      val r4 = '{summon[${T} <:< Any]} // error
+    }
+
+    {
+      val s = '{Option.empty[${T}]}
+      val r = '{identity($s)} // works
+      val r2 = '{identity(${s: Expr[Option[T]]})} // error // error
```

Comment:
It's not obvious why this is not accepted by the check.

Reply:
```
32 |      val r2 = '{identity[Option[T]](${s: Expr[Option[T]]})} // error // error
   |                                     ^^^^^^^^^^^^^^^^^^^^^
   |    (Test.this.given_Type_T : => quoted.Type[Test.this.T]) is not stable
```

All these can be fixed by making the `given`  stable

```scala
val getT: Type[T] = T // need this to avoid getting `null`
given getT.type = getT
```

Reply:
Added an extra test for it

Reply:
Why it has to be stable?

Reply:
Type splices require a stable prefix to be correctly encoded. It also avoids having to deal with side effects in the transformation performed in `ReifyQuotes`. Most code receives the quoted type as a parameter and hence is stable.

Reply:
It's a restriction in TASTy? This is unprecedented in Scala, as usually only the path in types needs to be stable.

Reply:
This is a path in a type. A type that is inferred and then healed.